iidm/iidm-scripting/src/main/groovy/com/powsybl/iidm/network/scripting/IdentifiableExtension.groovy

```diff
@@ -9,23 +9,58 @@ package com.powsybl.iidm.network.scripting
 import com.powsybl.commons.PowsyblException
 import com.powsybl.commons.extensions.Extension
 import com.powsybl.iidm.network.Identifiable
+import com.powsybl.iidm.network.util.Properties
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
 class IdentifiableExtension {
-
     static Object propertyMissing(Identifiable self, String name) {
         // first check if an extension exist then a property
         Extension extension = self.getExtensionByName(name)
-        extension != null ? extension : self.properties[name]
+        if (extension != null) {
+            extension
+        } else {
+            switch(self.getPropertyType(name)) {
+                case Properties.Type.BOOLEAN:
+                    self.getProperty(name).orElse(null)
+                    break;
+                case Properties.Type.DOUBLE:
+                    if (self.getProperty(name).isPresent()) {
+                        self.getProperty(name).get()
+                    }
+                    break;
+                case Properties.Type.INTEGER:
+                    if (self.getProperty(name).isPresent()) {
+                        self.getProperty(name).get()
+                    }
+                    break;
+                default:
+                    self.getProperty(name).orElse(null)
+            }
+        }
     }
-
     static void propertyMissing(Identifiable self, String name, Object value) {
```

Comment:
The implementation of this method is not correct: the property is created not get. Could you create a unit test that demonstrate the bug you make, and fix it?